src/server/api/routers/rooms.ts

```diff
@@ -64,4 +64,23 @@ export const roomsRouter = createTRPCRouter({
       const [res] = await ctx.db.select().from(rooms).where(eq(rooms.id, input.id)).limit(1);
       return res;
     }),
+  delete: teacherProcedure
+    .input(
+      z.object({
+        id: z.string()
+      }),
+    )
+    .mutation( async ({ ctx, input }) => {
+      const checkClasses = await ctx.db.select().from(classes).where(eq(classes.roomId, input.id));
+
+      if (!checkClasses) {
+        const res = await ctx.db
+          .delete(rooms)
+          .where(eq(rooms.id, input.id))
+          .returning();
+        return res;
+      } else {
+        return ("You can't delete this room as there are classes scheduled.")
+      }
```

Comment:
We could use an early return here which would make this slightly more readable. [More info](https://dev.to/jpswade/return-early-12o5#:~:text=An%20early%20return%2C%20or%20%E2%80%9Creturn,the%20beginning%20of%20a%20method.)

```suggestion
      if (checkClasses) {
return ("You can't delete this room as there are classes scheduled.")
      }         
      const res = await ctx.db
          .delete(rooms)
          .where(eq(rooms.id, input.id))
          .returning();
        return res;
```

Reply:
the "if (!checkClasses)" check didn't work for me, I had to change it to "if (checkClasses.length === 0)". I added all the other suggestions into the new commit.